tests/test_csc.py

```diff
@@ -0,0 +1,302 @@
+            await self.check_exposureState(harness.remote, ExposureState.INTEGRATING)
+            # the exposure state should be CANCELLED after a cancelled exposure
+            await self.check_exposureState(harness.remote, ExposureState.CANCELLED)
+
+    async def test_simulator(self):
+        """Test that we can turn the simulation mode on and off.
+
+        We know we are in simulation mode if the device "library" loaded by
+        the device is a different mock than `self.patch`.
+        """
+        async with Harness(initial_state=salobj.State.STANDBY) as harness:
+            # Check that we are properly in STANDBY at the start
+            await self.check_summaryState(harness.remote, salobj.State.STANDBY)
+            self.assertIsNone(harness.csc.device)
+
+            setsm_data = harness.remote.cmd_setSimulationMode.DataType()
+            setsm_data.mode = 1
+            await harness.remote.cmd_setSimulationMode.start(setsm_data, timeout=STD_TIMEOUT)
+            # nothing should happen until we switch to DISABLED
+            self.assertIsNone(harness.csc.device)
+
+            await harness.remote.cmd_start.start(timeout=STD_TIMEOUT)
+            await self.check_summaryState(harness.remote, salobj.State.DISABLED)
+            # The CSC's internal simulator should be different from our mock.
+            self.assertNotEqual(self.patch.return_value, harness.csc.device.libavs)
+
+            # It should remain different in ENABLED, and back to DISABLED
+            await harness.remote.cmd_enable.start(timeout=STD_TIMEOUT)
+            await self.check_summaryState(harness.remote, salobj.State.ENABLED)
+            self.assertNotEqual(self.patch.return_value, harness.csc.device.libavs)
+            await harness.remote.cmd_disable.start(timeout=STD_TIMEOUT)
+            await self.check_summaryState(harness.remote, salobj.State.DISABLED)
+            self.assertNotEqual(self.patch.return_value, harness.csc.device.libavs)
+
+            # in STANDBY, there should be no connected device
+            await harness.remote.cmd_standby.start(timeout=STD_TIMEOUT)
+            await self.check_summaryState(harness.remote, salobj.State.STANDBY)
+            self.assertIsNone(harness.csc.device)
+
+            # Turning the simulator off should give us the test patch again.
+            setsm_data = harness.remote.cmd_setSimulationMode.DataType()
+            setsm_data.mode = 0
+            await harness.remote.cmd_setSimulationMode.start(setsm_data, timeout=STD_TIMEOUT)
+            self.assertIsNone(harness.csc.device)
+            await harness.remote.cmd_start.start(timeout=STD_TIMEOUT)
+            await self.check_summaryState(harness.remote, salobj.State.DISABLED)
+            self.assertEqual(self.patch.return_value, harness.csc.device.libavs)
+
+    async def test_telemetry(self):
+        """Test that telemetry is emitted in the correct states.
+
+        NOTE: if the CSC does not close the telemetry loop when the CSC is
```

Comment:
Pick but...consider a standard sphinx Notes section

Reply:
I don't think test docstrings ever get built into final documentation.

Reply:
Why not use Sphinx doc format everywhere? That's what I do. Makes it easier for anyone reading the code.